pkg/apis/pipeline/v1beta1/run_interface.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+)
+
+// RunsToCompletionStatus is implemented by TaskRun.Status and PipelineRun.Status
+type RunsToCompletionStatus interface {
+	GetCondition(t apis.ConditionType) *apis.Condition
+	InitializeConditions()
+	SetCondition(newCond *apis.Condition)
+}
+
+// RunsToCompletion is implemented by TaskRun and PipelineRun
+type RunsToCompletion interface {
+	GetTypeMeta() *metav1.TypeMeta
+	GetObjectMeta() *metav1.ObjectMeta
+	GetOwnerReference() metav1.OwnerReference
+	GetStatus() RunsToCompletionStatus
+	IsDone() bool
+	HasStarted() bool
+	IsCancelled() bool
+	HasTimedOut() bool
+	GetRunKey() string
+}
```

Comment:
@afrittoli any reason for defining those in here and not where it is used ? (it, then, wouldn't need to be exported even :angel:)

Reply:
Right now it is only used by events, but I was hoping in future to use this to generalise more functions, perhaps even have a base reconciler for `RunToCompletion`.

That would help if we wanted to have dedicated reconcilers for metrics, cloud events or else in the future.

We could have this private to start with and export it in the next PRs if you preferred so.

Since I was adding functions to pipeline run and taskrun types to match this interface, I thought it would make sense to have the interface in the same place, but we could also move it to the controller package.

Reply:
> We could have this private to start with and export it in the next PRs if you preferred so.

That would be my *preferred* approach indeed :stuck_out_tongue_closed_eyes: I definitely prefer more unexported interfaces (to use the power of the *implicit* part of the interfaces) and small exported interfaces.

Reply:
done